lib/index.ts

```diff
@@ -1,147 +1,147 @@
-                if (!err && user) {
-                    next()
-                } else {
-                    res.status(401).end()
-                }
-            })(req, res, next)
-        })
-    }
-
-    // configure heartbeat if needed
-    if (!optIsFalsey(opts.heartbeatEnabled)) {
-        const heartbeatRouter = heartbeatRouterCreator({
-            peerList: peerList,
-            timeoutPeriod: opts.heartbeatMs || 30 * 1000,
-            gcInterval: opts.heartbeatGcMs || 15 * 1000
-        })
-
-        // use the heartbeat peerList instead
-        peerList = heartbeatRouter.peerList
-
-        app.use(heartbeatRouter)
-    }
-
-    // configure capacity if needed
-    if (!optIsFalsey(opts.capacityEnabled)) {
-        const capacityRouter = capacityRouterCreator({
-            peerList: peerList
-        })
-
-        // use the capacity peer list instead
-        peerList = capacityRouter.peerList
-
-        app.use(capacityRouter)
-    }
-
-    // configure peer recognition router if needed
-    if (!optIsFalsey(opts.recognitionEnabled)) {
-        const recognitionRouter = recognitionRouterCreator({
-            peerList: peerList
-        })
-
-        // use the recognition peer list instead
-        peerList = recognitionRouter.peerList
-
-        app.use(recognitionRouter)
-    }
-
-    // configure the webrtc-signal-http router
-    const signalRouter = signalRouterCreator({...opts,
-        // we use a predefined peerList in case we need to support heartbeats
-        peerList: peerList,
-
-        // see above, we use a different logger to capture more scenarios
-        enableLogging: false
-    })
-
-    // configure webrtc-signal-http-publisher if needed
-    if (!optIsFalsey(opts.publishState)) {
-        app._publisher = new Publisher(signalRouter)
-    }
-
-    // use webrtc-signal-http router
-    app.use(signalRouter)
-
-    // expose the peerList on the app
-    app.peerList = peerList
-
-    // return a configured app, ready to be used
-    return app
-}
+import * as express from "express";
+import * as expressBunyan from "express-bunyan-logger";
+import * as passport from "passport";
+import { ITokenPayload, BearerStrategy, VerifyCallback, IBearerStrategyOption } from "passport-azure-ad";
+import { signalRouterCreator, IPeerRequest, PeerList } from "webrtc-signal-http-ts";
+import { ISignalerOpts, IPeerBearerStrategyOpts, IExpressApp } from "./modules";
+const heartbeatRouterCreator = require("webrtc-signal-http-heartbeat");
+const capacityRouterCreator = require("webrtc-signal-http-capacity");
+const recognitionRouterCreator = require("webrtc-signal-http-peer-identification");
+const Publisher = require("webrtc-signal-http-publisher");
+
+const optIsFalsey = (opt: string | boolean) => {
+    return !opt ||
+        opt === 'false' ||
+        (typeof(opt) === 'string' && opt.toLowerCase() === 'false') 
+}
+
+module.exports = (opts: ISignalerOpts) => {
+    const app = express() as IExpressApp; 
```

Comment:
these lines end with `;` but the rest of the file does not. We should be consistent

Reply:
Is there a preference besides consistency?
removed the two semi colon lines

Reply:
My preference would be to add `;` on all lines, but this is more for consistency

Reply:
done and added a linter!